metron-platform/metron-elasticsearch/src/test/java/org/apache/metron/elasticsearch/integration/ElasticsearchSearchIntegrationTest.java

```diff
@@ -97,45 +118,63 @@ protected static InMemoryComponent startIndex() throws Exception {
-    for (Object o : broArray) {
-      JSONObject jsonObject = (JSONObject) o;
-      IndexRequestBuilder indexRequestBuilder = es.getClient()
-          .prepareIndex("bro_index_2017.01.01.01", "bro_doc");
-      indexRequestBuilder = indexRequestBuilder.setId((String) jsonObject.get("guid"));
-      indexRequestBuilder = indexRequestBuilder.setSource(jsonObject.toJSONString());
-      indexRequestBuilder = indexRequestBuilder
-          .setTimestamp(jsonObject.get("timestamp").toString());
-      bulkRequest.add(indexRequestBuilder);
+    es.getClient().admin().indices().prepareCreate(snortIndex)
+            .addMapping("snort_doc", JSONUtils.INSTANCE.toJSON(snortTemplate.get("mappings"), false)).get();
+
+    // setup the classes required to write the test data
+    AccessConfig accessConfig = createAccessConfig();
+    ElasticsearchClient client = ElasticsearchUtils.getClient(createGlobalConfig());
+    ElasticsearchRetrieveLatestDao retrieveLatestDao = new ElasticsearchRetrieveLatestDao(client);
+    ElasticsearchColumnMetadataDao columnMetadataDao = new ElasticsearchColumnMetadataDao(client);
+    ElasticsearchRequestSubmitter requestSubmitter = new ElasticsearchRequestSubmitter(client);
+    ElasticsearchUpdateDao updateDao = new ElasticsearchUpdateDao(client, accessConfig, retrieveLatestDao);
+    ElasticsearchSearchDao searchDao = new ElasticsearchSearchDao(client, accessConfig, columnMetadataDao, requestSubmitter);
+
+    // write the test documents for Bro
+    List<String> broDocuments = new ArrayList<>();
+    for (Object broObject: (JSONArray) new JSONParser().parse(broData)) {
+      broDocuments.add(((JSONObject) broObject).toJSONString());
     }
-    JSONArray snortArray = (JSONArray) new JSONParser().parse(snortData);
-    for (Object o : snortArray) {
-      JSONObject jsonObject = (JSONObject) o;
-      IndexRequestBuilder indexRequestBuilder = es.getClient()
-          .prepareIndex("snort_index_2017.01.01.02", "snort_doc");
-      indexRequestBuilder = indexRequestBuilder.setId((String) jsonObject.get("guid"));
-      indexRequestBuilder = indexRequestBuilder.setSource(jsonObject.toJSONString());
-      indexRequestBuilder = indexRequestBuilder
-          .setTimestamp(jsonObject.get("timestamp").toString());
-      bulkRequest.add(indexRequestBuilder);
+    es.add(updateDao, broIndex, "bro", broDocuments);
+
+    // write the test documents for Snort
+    List<String> snortDocuments = new ArrayList<>();
+    for (Object snortObject: (JSONArray) new JSONParser().parse(snortData)) {
+      snortDocuments.add(((JSONObject) snortObject).toJSONString());
     }
-    BulkResponse bulkResponse = bulkRequest.execute().actionGet();
-    if (bulkResponse.hasFailures()) {
-      throw new RuntimeException("Failed to index test data");
+    es.add(updateDao, snortIndex, "snort", snortDocuments);
+
+    // wait until the test documents are visible
+    assertEventually(() -> Assert.assertEquals(10, findAll(searchDao).getTotal()));
```

Comment:
Unless I added this to hold up the test and wait for the tests data to be 'visible', some of the tests would reliably fail because the test data wasn't loaded.

Reply:
@nickwallen is it possible to parameterize the call to do something similar to this? https://github.com/apache/metron/pull/1242/files#diff-99b67fd77500d6232462dc7c27ecff67R148

It's possible this is only available on batch calls, but I wanted to mention it in case we can get away with the assertEventually calls.

Reply:
I have tried to do exactly this on a private branch that is closer to the code base in #1269.  I made the integration tests supply a RefreshPolicy that was supposed to hold them up until the loaded data is discoverable/searchable.  I was never able to get that to work consistently though despite what the ES docs say.  I still needed the `assertEventually` for the integration tests to pass consistently.

Reply:
Are you able to get this to fail consistently? I ran it a number of times and couldn't get it to fail. I'm wondering if there are any other interesting entries in the logs when this occurs. These tests from ES might be of help - https://github.com/elastic/elasticsearch/pull/17986/files#diff-1c9f982dbd2f9ddb2853d135884621b5R112

Reply:
> Are you able to get this to fail consistently? I ran it a number of times and couldn't get it to fail.

Can you describe what you did? You just removed line 190 and the tests continue to work reliably?

Reply:
I meant on the es client change PR, not on this PR. Yes, I had removed assertEventually calls and it consistently passed for me in the search integration test.